src/lib/utils/mem_ops.h

```diff
@@ -127,6 +176,20 @@ inline constexpr void copy_mem(T* out, const T* in, size_t n)
    }
 }
 
+/**
+* Copy memory
+* @param out the destination array
+* @param in the source array
+*/
+template <ranges::contiguous_output_range OutR, std::ranges::contiguous_range InR>
+inline constexpr void copy_mem(OutR&& out, InR&& in)
+   requires std::is_same_v<std::ranges::range_value_t<OutR>, std::ranges::range_value_t<InR>> &&
+            std::is_trivially_copyable_v<std::ranges::range_value_t<InR>>
+{
+   ranges::assert_equal_byte_lengths(out, in);
```

Comment:
If both `in` and `out` have static extents (e.g. static-extent `std::span<>` and/or `std::array<>`), `assert_equal_byte_lengths` will only verify the extents statically. There won't be another check during runtime.